app/models/booking_response.rb

```diff
@@ -168,4 +167,26 @@ def not_allowed_visitor_ids
     @not_allowed_visitor_ids ||=
       unlisted_visitor_ids + banned_visitor_ids
   end
+
+  def no_banned_or_unlisted_visitors?
```

Comment:
I agree, using the validation callbacks to calculate additional rejection reasons is a bit strange and is a smell in this class. I think we can decouple the calculation of automated rejection reasons from the class into a different class perhaps.